library/src/main/kotlin/org/contentauth/c2pa/manifest/ManifestBuilder.kt

```diff
@@ -0,0 +1,260 @@
+package org.contentauth.c2pa.manifest
+
+import org.json.JSONArray
+import org.json.JSONObject
+import java.util.*
+
+data class ClaimGenerator(
+    val name: String,
+    val version: String,
+    val icon: String? = null
+)
+
+data class Ingredient(
+    val title: String? = null,
+    val format: String,
+    val instanceId: String = UUID.randomUUID().toString(),
+    val documentId: String? = null,
+    val provenance: String? = null,
+    val hash: String? = null,
+    val relationship: String = "parentOf",
+    val validationStatus: List<String> = emptyList(),
+    val thumbnail: Thumbnail? = null
+)
+
+data class Thumbnail(
+    val format: String,
+    val identifier: String,
+    val contentType: String = "image/jpeg"
```

Comment:
How come the format and contentType are separate fields? I also don't think we should default it to JPEG. The SDK will infer the format from the identifier and will auto-generate PNG, JPEG, or other formats depending on the circumstance.